gtsam/linear/NoiseModel.cpp

```diff
@@ -654,6 +654,13 @@ void Robust::WhitenSystem(Matrix& A1, Matrix& A2, Matrix& A3, Vector& b) const{
   robust_->reweight(A1,A2,A3,b);
 }
 
+Robust::shared_ptr Robust::Create(const RobustModel::shared_ptr& robust,
```

Comment:
Theoretically, the noisemodel inside robust should be Gaussian. However, the old implementation uses a general noisemodel. After changing it to Gaussian, I added this constructor to fit the old API.

Reply:
I don't understand, if it is not `Gaussian` then this conversion will 100% fail.

Reply:
I think this complicated matters rather than simplifies them. You can’t change the old API, anyway without a version bump. Please undo the change and throw an exception if not (subclass of) Gaussian - and remove this new function.

Reply:
It must be Gaussian @ProfFan 
I've revised by Frank's suggestion in the new commit.